specification/attestation.adoc

```diff
@@ -1,82 +1,245 @@
+measurements for those elements could be shared with e.g. a measured boot
+attestation.
+2. The TSM TCB elements are the confidential compute specific but TVM agnostic
+elements for the overall TCB.
+3. The TVM TCB is composed of the TVM specific elements. 
+
+The TCB elements for each of them is summarized in the following table:
+
+.AP-TEE Workload TCB Elements
+[cols="^1, <5, <4"]
+|===
+|TCB |Hardware Elements |Software Elements
+
+ | Platform
+a| * HW RoT for boot, measurement and storage
+   * All CPU hardware logic, including MMU and caches
+   * All SoC subsystems, including memory confidentiality, integrity and replay-protection for volatile memory
+   * IOMMU and translation agents
+a| * All M-mode firmwares, except for the TSM-driver
+
+ | TSM
+ | N/A
+a| * TEE Security Manager (TSM) and its U-mode components
+   * TSM-Driver that hosts a TEEI (with TH-ABI and TG-ABI security routines). 
+
+ | TVM
+a| * Directly assigned, TEE-IO compliant devices
+ | * All TVM measured pages
+|===
+
+==== TVM Measurements
+
+At a high level, a TVM measurement is separated into a set of static
+measurements and a dynamic one.
+
+===== Static Measurements
+
+TVM static measurements are generated from the AP-TEE workload TCB elements
+involved in the TVM construction. Any TCB element that directly or indirectly
+supports a TVM generation must be measured into the TVM static measurement
+registers. Once a TVM is finalized, i.e. after the
+`sbi_tee_host_finalize_tvm()` TH-ABI is called, the TVM static measurements
+must no longer be extended.
+
+Static measurements are stored in dedicated measurement registers and an AP-TEE
+implementation must provide at least 1 and at most 16 of them. Some of the
+static measurement registers cover platform or TSM specific TCB elements while
+other are measuring TVM specific TCB pieces, and have respectively platform
+bound and TVM bound lifecycles. As such the former may be stored in the RoT
+for Storage while the latter should be maintained and temporarily stored in TSM
+confidential memory, effectively making the TSM an extended root of trust of
+measurements for the TVM.
+
+Which and how TCB elements are measured and stored into the static measurement
+registers is implementation specific. One possible layout, based on a 6 static
+measurement registers AP-TEE implementation may look like the following table:
+
+.Static Measurements Layout Example (6 registers)
+[cols="^1,<3,<5,<1"]
+|===
+|Register |Usage |TCB Elements |Storage
+ | 0
+ | Platform Firmware
+a| * RoT firmware
+   * All SoC subsystem firmwares
+   * M-mode firmware
+ | RoT
+
+ | 1
+ | Platform Configuration
+ | All firmware manifests
+ | RoT
+
+ | 2
+ | TSM Software
+a| * TSM-Driver
+   * TSM
+ | RoT
+
+ | 3
+ | TSM Configuration
+a| * TSM-Driver manifest
+   * TSM manifest
+ | RoT
+
+ | 4
+ | TVM Code
+ | TVM measured pages
+ | TSM
+
+ | 5
+ | TVM Configuration
+ | TVM EPC and initial arguments
```

Comment:
No earlier reference for this acronym EPC.

Reply:
Fixed now.